solidserver/provider.go

```diff
@@ -58,6 +61,14 @@ func Provider() *schema.Provider {
 				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^([0-9]\.[0-9]\.[0-9]((\.[pP]\d+[a-z]?)|[a-z])?)?$`), "Invalid Version Number"),
 				Description:  "SOLIDServer Version in case API user does not have admin permissions",
 			},
+			"proxy_url": {
+				Type:             schema.TypeString,
+				Required:         false,
+				Optional:         true,
+				DefaultFunc:      schema.EnvDefaultFunc("SOLIDServer_PROXY_URL", ""),
```

Comment:
I thought it was better to have a specific env variable, rather than reusing the generic `https_proxy`/`HTTPS_PROXY`. As existing uses may have these set in their environment, and see unexpected breakage if the provider began to use them and they were not expecting a proxy to be used (it would be needed to implement e.g. `no_proxy` logic too)

And it is probably better to have an explicit config option anyway. As it is trivial for a user who wants to reuse e.g. `https_proxy` like:
```
SOLIDServer_PROXY_URL=$https_proxy terraform plan ...
```